File: core/cpp/src/test/vx/ui/html/uihtml_test.cpp
#include "../../../../main/vx/ui/html/uihtml.hpp"
#include "../../../../main/vx/core.hpp"
#include "../../../../main/vx/test.hpp"
#include "../../../../main/vx/data/file.hpp"
#include "../../../../main/vx/web/html.hpp"
#include "../../../../main/vx/web/htmldoc.hpp"
#include "../../../../main/vx/type.hpp"
#include "../../../../main/vx/ui/ui.hpp"
#include "uihtml_test.hpp"

namespace vx_ui_html_uihtml_test {


  vx_test::Type_testcaselist test_cases(vx_core::Type_context context) {
    vx_core::vx_Type_listany listtestcase;
    vx_test::Type_testcaselist output = vx_core::vx_any_from_any(
      vx_test::t_testcaselist,
      vx_test::t_testcaselist->vx_new_from_list(listtestcase)
    );
    return output;
  }

  vx_test::Type_testcoveragesummary test_coveragesummary() {
    vx_test::Type_testcoveragesummary output = vx_core::vx_new(vx_test::t_testcoveragesummary, {
      vx_core::vx_new_string(":testpkg"), vx_core::vx_new_string("vx/ui/html/uihtml"),
      vx_core::vx_new_string(":constnums"), vx_core::vx_new(vx_test::t_testcoveragenums, {
        vx_core::vx_new_string(":pct"), vx_core::vx_new_int(0), 
        vx_core::vx_new_string(":tests"), vx_core::vx_new_int(0), 
        vx_core::vx_new_string(":total"), vx_core::vx_new_int(7)
      }),
      vx_core::vx_new_string(":docnums"), vx_core::vx_new(vx_test::t_testcoveragenums, {
        vx_core::vx_new_string(":pct"), vx_core::vx_new_int(60), 
        vx_core::vx_new_string(":tests"), vx_core::vx_new_int(23), 
        vx_core::vx_new_string(":total"), vx_core::vx_new_int(38)
      }),
      vx_core::vx_new_string(":funcnums"), vx_core::vx_new(vx_test::t_testcoveragenums, {
        vx_core::vx_new_string(":pct"), vx_core::vx_new_int(0), 
        vx_core::vx_new_string(":tests"), vx_core::vx_new_int(0), 
        vx_core::vx_new_string(":total"), vx_core::vx_new_int(30)
      }),
      vx_core::vx_new_string(":bigospacenums"), vx_core::vx_new(vx_test::t_testcoveragenums, {
        vx_core::vx_new_string(":pct"), vx_core::vx_new_int(0), 
        vx_core::vx_new_string(":tests"), vx_core::vx_new_int(0), 
        vx_core::vx_new_string(":total"), vx_core::vx_new_int(31)
      }),
      vx_core::vx_new_string(":bigotimenums"), vx_core::vx_new(vx_test::t_testcoveragenums, {
        vx_core::vx_new_string(":pct"), vx_core::vx_new_int(0), 
        vx_core::vx_new_string(":tests"), vx_core::vx_new_int(0), 
        vx_core::vx_new_string(":total"), vx_core::vx_new_int(31)
      }),
      vx_core::vx_new_string(":totalnums"), vx_core::vx_new(vx_test::t_testcoveragenums, {
        vx_core::vx_new_string(":pct"), vx_core::vx_new_int(0), 
        vx_core::vx_new_string(":tests"), vx_core::vx_new_int(0), 
        vx_core::vx_new_string(":total"), vx_core::vx_new_int(37)
      }),
      vx_core::vx_new_string(":typenums"), vx_core::vx_new(vx_test::t_testcoveragenums, {
        vx_core::vx_new_string(":pct"), vx_core::vx_new_int(100), 
        vx_core::vx_new_string(":tests"), vx_core::vx_new_int(0), 
        vx_core::vx_new_string(":total"), vx_core::vx_new_int(0)
      })
    });
    return output;
  }

  vx_test::Type_testcoveragedetail test_coveragedetail() {
    vx_test::Type_testcoveragedetail output = vx_core::vx_new(vx_test::t_testcoveragedetail, {
      vx_core::vx_new_string(":testpkg"), vx_core::vx_new_string("vx/ui/html/uihtml"),
      vx_core::vx_new_string(":typemap"), vx_core::vx_new(vx_core::t_intmap, {
        
      }),
      vx_core::vx_new_string(":constmap"), vx_core::vx_new(vx_core::t_intmap, {
        vx_core::vx_new_string(":layout-app-html"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":layout-else-html"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":layout-image-html"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":layout-label-html"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":layoutenginehtml"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":style-hidden"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":style-selected"), vx_core::vx_new_int(0)
      }),
      vx_core::vx_new_string(":funcmap"), vx_core::vx_new(vx_core::t_intmap, {
        vx_core::vx_new_string(":boolean-print-html"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":boolean-writeeventsall<-ui"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":boolean-writeeventsall<-uimap"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":context-write"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":divchild<-ui"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":divchildlist<-uimap"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":divchildlist<-uimap_1"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":divchildlist<-uimap-origmap-parent"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":node-app<-ui-orig-parent"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":node-default<-ui-orig-parent"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":node-image<-ui-orig-parent"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":node-label<-ui-orig-parent"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":node-layout<-node-ui"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":node<-ui-orig-parent"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":string-style<-font"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":string-style<-image"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":string-stylename<-name-styletype"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":style<-fontface"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":style<-style"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":stylelist-extra<-ui"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":stylelist-reset"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":stylelist<-fontfacemap"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":stylelist<-stylelist"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":stylelist<-stylemap"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":stylemap<-stylemap"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":stylesheet-layout-html"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":stylesheet<-stylesheet"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":ui-layout-app<-ui-orig-parent"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":ui-layout-default<-ui-orig-parent"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":ui-layout-image<-ui-orig-parent"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":ui-layout-label<-ui-orig-parent"), vx_core::vx_new_int(0)
      })
    });
    return output;
  }

  vx_test::Type_testpackage test_package(vx_core::Type_context context) {
    vx_test::Type_testcaselist testcaselist = vx_ui_html_uihtml_test::test_cases(context);
    vx_test::Type_testcoveragesummary testcoveragesummary = vx_ui_html_uihtml_test::test_coveragesummary();
    vx_test::Type_testcoveragedetail testcoveragedetail = vx_ui_html_uihtml_test::test_coveragedetail();
    vx_test::Type_testpackage output = vx_core::vx_new(vx_test::t_testpackage, {
      vx_core::vx_new_string(":testpkg"), vx_core::vx_new_string("vx/ui/html/uihtml"),
      vx_core::vx_new_string(":caselist"), testcaselist,
      vx_core::vx_new_string(":coveragesummary"), testcoveragesummary,
      vx_core::vx_new_string(":coveragedetail"), testcoveragedetail
    });
    return output;
  }

}

File: core/js/test/vx/ui/html/uihtml_test.js
'strict mode'

import vx_ui_html_uihtml from "../../../../src/vx/ui/html/uihtml.js"
import vx_core from "../../../../src/vx/core.js"
import vx_test from "../../../../src/vx/test.js"
import vx_data_file from "../../../../src/vx/data/file.js"
import vx_web_html from "../../../../src/vx/web/html.js"
import vx_web_htmldoc from "../../../../src/vx/web/htmldoc.js"
import vx_type from "../../../../src/vx/type.js"
import vx_ui_ui from "../../../../src/vx/ui/ui.js"

export default class vx_ui_html_uihtml_test {


  static test_package(context) {
    const testcaselist = vx_ui_html_uihtml_test.test_cases(context)
    const output = vx_core.f_new(
      vx_test.t_testpackage,
      ":testpkg", "vx/ui/html/uihtml",
      ":caselist", testcaselist,
      ":coveragesummary", vx_ui_html_uihtml_test.test_coveragesummary(),
      ":coveragedetail", vx_ui_html_uihtml_test.test_coveragedetail()
    );
    return output;
  }

  static test_coveragesummary() {
    return vx_core.f_new(
      vx_test.t_testcoveragesummary,
      "testpkg",   "vx/ui/html/uihtml", 
      "constnums", vx_core.f_new(vx_test.t_testcoveragenums, ":pct", 0, ":tests", 0, ":total", 7), 
      "docnums", vx_core.f_new(vx_test.t_testcoveragenums, ":pct", 66, ":tests", 30, ":total", 45), 
      "funcnums", vx_core.f_new(vx_test.t_testcoveragenums, ":pct", 0, ":tests", 0, ":total", 37), 
      "bigospacenums", vx_core.f_new(vx_test.t_testcoveragenums, ":pct", 100, ":tests", 0, ":total", 0), 
      "bigotimenums", vx_core.f_new(vx_test.t_testcoveragenums, ":pct", 100, ":tests", 0, ":total", 0), 
      "totalnums", vx_core.f_new(vx_test.t_testcoveragenums, ":pct", 0, ":tests", 0, ":total", 44), 
      "typenums", vx_core.f_new(vx_test.t_testcoveragenums, ":pct", 100, ":tests", 0, ":total", 0)
    )
  }

  static test_coveragedetail() {
    return vx_core.f_new(
      vx_test.t_testcoveragedetail,
      "testpkg", "vx/ui/html/uihtml",
      "typemap",
        vx_core.f_new(
          vx_core.t_intmap
        ),
      "constmap",
        vx_core.f_new(
          vx_core.t_intmap,
          "layout-app-html", 0,
          "layout-else-html", 0,
          "layout-image-html", 0,
          "layout-label-html", 0,
          "layoutenginehtml", 0,
          "style-hidden", 0,
          "style-selected", 0
        ),
      "funcmap",
        vx_core.f_new(
          vx_core.t_intmap,
          "boolean-layoutremove-html", 0,
          "boolean-layoutselected-html", 0,
          "boolean-layoutvisible-html", 0,
          "boolean-print-html", 0,
          "boolean-writeclass<-ui", 0,
          "boolean-writeeventsall<-ui", 0,
          "boolean-writeeventsall<-uimap", 0,
          "boolean-writeselected<-ui", 0,
          "boolean-writevisible<-ui", 0,
          "context-write", 0,
          "divchild<-ui", 0,
          "divchildlist<-uimap", 0,
          "divchildlist<-uimap_1", 0,
          "divchildlist<-uimap-origmap-parent", 0,
          "node-app<-ui-orig-parent", 0,
          "node-default<-ui-orig-parent", 0,
          "node-image<-ui-orig-parent", 0,
          "node-label<-ui-orig-parent", 0,
          "node-layout<-node-ui-parent", 0,
          "node<-ui-orig-parent", 0,
          "string-class<-ui", 0,
          "string-style<-font", 0,
          "string-style<-image", 0,
          "string-stylename<-name-styletype", 0,
          "style<-fontface", 0,
          "style<-style", 0,
          "stylelist-extra<-ui", 0,
          "stylelist-reset", 0,
          "stylelist<-fontfacemap", 0,
          "stylelist<-stylelist", 0,
          "stylelist<-stylemap", 0,
          "stylemap<-stylemap", 0,
          "stylesheet-layout-html", 0,
          "stylesheet<-stylesheet", 0,
          "ui-layout-app<-ui-orig-parent", 0,
          "ui-layout-default<-ui-orig-parent", 0,
          "ui-layout-image<-ui-orig-parent", 0,
          "ui-layout-label<-ui-orig-parent", 0
        )
    )
  }

  static test_cases(context) {
    const output = vx_core.f_new(
      vx_test.t_testcaselist
    )
    return output
  }
}
